extensions/http/src/main/java/jolie/net/HttpProtocol.java

```diff
@@ -1298,9 +1374,72 @@ private void recv_checkReceivingOperation( HttpMessage message, DecodedMessage d
 				}
 			}
 		}
+	}
+
+	private void recv_extractReceivingOperation( HttpMessage message, DecodedMessage decodedMessage ) {
+		if( message.getMethod().isEmpty() )
+			return;
+
+		Value configurationValue = getParameterFirstValue( CommProtocol.Parameters.OPERATION_SPECIFIC_CONFIGURATION );
+		Iterator< Entry< String, ValueVector > > configurationIterator =
+			configurationValue.children().entrySet().iterator();
+		boolean foundMatch = false;
+		while( configurationIterator.hasNext() & !foundMatch ) {
+			Entry< String, ValueVector > configEntry = configurationIterator.next();
+
+			Value opConfig = configEntry.getValue().get( 0 );
+			Value uriTemplateResult = Value.create();
+			String uri = message.requestPath().substring( 0, 2 ).equals( "//" ) // FIXME, TODO: strange jolie
+				// double slash
+				? message.requestPath().substring( 1 )
+				: message.requestPath();
+			if( opConfig.hasChildren( Parameters.TEMPLATE ) ) {
+				uriTemplateResult =
+					UriUtils.match( opConfig.getFirstChild( Parameters.TEMPLATE ).strValue(), uri );
+			}
+			String opConfigMethod = opConfig.getFirstChild( Parameters.METHOD ).strValue();
+
+			if( uriTemplateResult.boolValue() && message.getMethod().equalsIgnoreCase( opConfigMethod ) ) {
+				foundMatch = true;
+				decodedMessage.operationName = configEntry.getKey();
+				decodedMessage.resourcePath = "/";
+
+				String messagePath = "/".concat( configEntry.getKey() );
+				String paramString = "";
```

Comment:
I suggest using a `StringBuilder` here, and in any other place where you end up concatenating a lot of strings.